### PR TITLE
Features/spellcheck linting

### PR DESCRIPTION
**Pull request by shannonwells**
Problem
=======
We need spellchecking and a markdown linter.
[fixes #176963069](https://www.pivotaltracker.com/story/show/12345678)

markdown-spellcheck uses a `.spelling` file that contains custom words for your repo. You can add new words by editing it, adding one word (case sensitive) per line, or by running the new script, `npm run spellfix`

NOTE: I did not include a "lint-fix" equivalent because remark does not have a markdown code rewrite option without adding another package and writing a new script.

Solution
========
1. Add remark-lint, remark and markdown-spellcheck
2. Add scripts that perform linting and spellchecking, including an interactive one.
3. Use in the "lint" package.json script.
Correct any warnings/errors that come from running lint.

Steps to Verify:
----------------
```
npm i
npm run lint
```
The packages should install without dependency conflicts, the spellchecker should run, and the markdown linter should run with no errors or warnings.

You can also test the interactive spellchecking, which allows you to add items to the dictionary, the `.spelling` file, by misspelling something and then running
`npm run spellfix`

Use up-down arrow to select a fix type, such as adding to the dictionary or entering a new spelling.  It should fix the spelling for you or add a new word to the dictionary (for example).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,17 +13,35 @@
...
+        "spellcheck": "mdspell './pages/**/*.md' 'CODEOFCONDUCT.md'  --en-US --ignore-numbers --report",
+        "spellfix": "mdspell './pages/**/*.md' './*.md' '!./LICENSE.md'  --en-US --ignore-numbers",
+        "lint-md": "remark ./pages/**/*.md",
+        "lint": "npm run lint-md && npm run spellcheck"
     },
     "dependencies": {
         "@emotion/react": "^11.1.1",
         "@emotion/styled": "^11.0.0",
         "docz": "^2.3.1",
+        "markdown-spellcheck": "^1.3.0",
```
Comment (wilwade) [inline package.json:25]: Should these be devDependencies?

File: .spelling [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+EncryptionKeyList
+messageID
+dsnpData
+inReplyTo
+PrivateGraph
+PrivateGraphChange
+EncryptedInbox
+PrivateBroadcast
+spec
+Dokz
```
Comment (wilwade) [inline .spelling:38]: I feel like Dokz shouldn't be anywhere that counts as spell checking, but perhaps not?

File: .spelling [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+dsnpType
+fromAddress
+bytes20
+uri
+deadDropID
+DeadDrops
+actionType
+PrivateGraphKeyList
+toAddress
+utf-8
```
Comment (wilwade) [inline .spelling:56]: UTF-8?

**Review by rosalinekarr — APPROVED** [review]
LGTM. Maybe add a note about `spellfix` to the README?

**Review by wilwade — COMMENTED** [review]
I think the only thing I would like to see would be the devDeps. Other things are just comments.

File: package.json (hunk 1 above)
Comment (shannonwells) [reply package.json:25]: Yes they should. I moved them to deps b/c I was trying to debug the ci failure.  I didn't see this before Rose's approval. I'll file a follow up.

File: .spelling (hunk 2 above)
Comment (shannonwells) [reply .spelling:38]: it's probably irrelevant now since it's Docz

File: .spelling (hunk 3 above)
Comment (shannonwells) [reply .spelling:56]: I believe this one is case insensitive. I think it treats lowercase as case-insensitive. I should have noted that.

File: .spelling (hunk 2 above)
Comment (shannonwells) [reply .spelling:38]: I misspoke.  Docz is in the README, which should still be spellchecked.

**Pull request merged**